News grid

demi-johnson opened the pull request:
My first big stab at creating a template and markdown from scratch. I'm still trying to figure out how to make the text size of the intro text change according to the screen size.

demi-johnson committed:
Create news-grid.html

demi-johnson committed:
working on adding the cards

demi-johnson committed:
card edits

demi-johnson committed:
added card

demi-johnson committed:
added image into markdown

demi-johnson committed:
added grid and edited markdown

demi-johnson committed:
card edits

demi-johnson committed:
adding link

demi-johnson committed:
adding card markdown

demi-johnson committed:
figuring out the grid

trying to get the cards to line up next to each other

demi-johnson committed:
changed grid for cards

somehow got them to line up!

demi-johnson committed:
shifted cards down

demi-johnson committed:
adding card info and bringing in the header

demi-johnson committed:
edited the grid of header and intro

demi-johnson committed:
updating images

demi-johnson committed:
added images

demi-johnson committed:
set image height and width

demi-johnson committed:
trying to adjust mobile font size

demi-johnson committed:
adjusting font size

pglevy commented on _layouts/news-grid.html:
```diff
@@ -0,0 +1,63 @@
+      <section class="usa-card__header bg-blue-80v">
+            <div class="grid-container">
+                <h1 class="font-heading-3xl text-accent-cool">{{ page.title }}</h1>
+                <h2 class="font-alt-xl text-base-lightest tablet:grid-col-9">{{ page.description }}</h2>
```
This is functioning more as a subtitle than a heading. In that case, you can just mark up as a paragraph and uses classes to get the appearance you want. In general, we don't want to headings to style appearance alone.

demi-johnson replied:
In the USWDS font size and family utility they have type-based and role-based size utilities. What are the differences between those two? For this paragraph I used the role-based class, font-alt-xl.

pglevy commented on _layouts/news-grid.html:
```diff
@@ -0,0 +1,63 @@
+ <li class="tablet:grid-col-4 usa-card">
+    <div class="usa-card__container">
+      <header class="usa-card__header">
+        <h6 class="text-base-dark">{{ article.source }}</h6>
```
Similar issue to subtitle above but for additional reason: headings should follow their hierarchical order (2s after 1s, 3s after 2s, etc.). So we don't want to have an h6 followed by an h2. This is a common problem use case, where you have this kind of subtitle above the main title. I'll see if I can find a discussion of the preferred approach, but you can't go wrong with a paragraph and then use classes for styling to get the appearance you want.

demi-johnson replied:
Okay thanks for explaining! This leads to some q's I have about using text styles in Figma and what happens during the handoff to developers. Definitely something we can meet and discuss because I think I could be more clear asking about it verbally, but I wanted to record it here so I don't forget.

pglevy commented on pages/news-grid.md:
```diff
@@ -0,0 +1,55 @@
+title: News
+description: Card-based layout that displays recent news articles or case studies
+intro: 18F partners with federal agencies to improve the user experience of government services by helping them build and buy technology. If you're looking to implement a requirement, update a public-facing website, or digitize a process, 18F can work with you to build a product or craft and execute an effective agile acquisition strategy.
+contact-email:
```
Remember to enter something here. Otherwise nothing gets pulled into the page and it shows up as an empty link.

pglevy commented on _layouts/news-grid.html:
```diff
@@ -0,0 +1,63 @@
+        </p>
+      </div>
+      <div class="usa-card__footer">
+        <a href="{{ article.link-source }}" class="usa-link link-arrow-right">{{ article.link }}</a>
```
Something to consider for accessibility is to avoid sequential links with the same link text. (Because when you tab through, it just sounds like 'read more... read more... read more...' to someone using a screenreader. 

As an alternative, you could use link the card title instead. If you want to keep the customizable link text, I would suggest including some examples of how you might vary those instead of entering the same text for all of them.

pglevy reviewed (changes requested):
Really nice work overall Demi! Just a couple heading things and some missing content. Other than that, looks good.

demi-johnson committed:
Updated subheading to p

demi-johnson committed:
changed an h6 to a p

demi-johnson committed:
made the title the article links

- removed the 'read more' links
- made the card title the link to the article
- changed the appearance of the title to be not underlined and primary blue

demi-johnson committed:
working on hover

demi-johnson committed:
added usa-link text for hover

wanted the linked title text to change colors on hover. the usa-link class did just that!

demi-johnson committed:
removed read me link from md

and added an email

demi-johnson committed:
updated email

added mailto: in the html to bring up the email link
